File: SW1/4.cpp
#include <iostream>
int n;
using namespace std;
vector<int> input;
int ans = 0;
void recur(int idx, int count, bool arr[]){

    if(arr[idx]){
        ans = max(ans,count+1); return;
    }
    arr[idx] = true;
    int nextidx = (idx + input[idx]);
    if(nextidx<0 || nextidx>=n){
        ans = max(ans,count); return;
    }
    recur(nextidx,count+1,arr);

}
int main(){
    cin>>n;
    input.resize(n);
    for(int i=0; i<n; i++) cin>>input[i];
    bool arr[101]={};
    recur(0,0,arr);
   
    recur(1,0,arr);
    recur(2,0,arr);
    cout<<ans;
}

File: SW1/4.py
n= int(input())
chk=[False]*n
L=[*map(int, input().split())]
def stamp(k,cnt):
    #print(k,cnt)
    if not chk[k+L[k]] and 0<=k+L[k] and k+L[k]<n:
        chk[k+L[k]]= True
        return stamp(k+L[k],cnt+1)
    else:
        return cnt+1

ans=-1
for i in range(3):
    chk=[False]*n
    ans=max(stamp(i,1),ans)
print(ans)


'''
10
3 5 -1 -2 4 4 3 -2 -3 -2
'''